### PR TITLE
Add unit tests for the task controller

### LINKED ISSUE #30: Write unit tests for `JobReconciler` and `TaskReconciler`
Neither the `JobReconciler` nor the `TaskReconciler` have unit tests. Lets write some.

### PR DESCRIPTION
**Pull request by chrisdoherty4**
Depends on #46 

Add unit tests for the task controller.

Closes #30

### CODE REVIEW

File: controllers/job_controller_test.go [hunk 1]
```diff
@@ -0,0 +1,158 @@
+package controllers_test
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	. "github.com/onsi/gomega"
```
Comment (jacobweinstock) [inline controllers/job_controller_test.go:8]: I'm concerned with a couple things here.

- The dot import. _"Don't use the import . notation, which can simplify tests that must run outside the package they are testing, but should otherwise be avoided."_ --[ref](https://go.dev/doc/effective_go#:~:text=Don%27t%20use%20the%20import%20.%20notation%2C%20which%20can%20simplify%20tests%20that%20must%20run%20outside%20the%20package%20they%20are%20testing%2C%20but%20should%20otherwise%20be%20avoided.)
- The use of this assert library. _"Avoid the use of 'assert' libraries to help your tests"_ --[ref](https://github.com/golang/go/wiki/TestComments#assert-libraries) With this one, i see that `ginko` and `gomega` are already used a bunch through out the repo. 

I won't block on using the assert libraries but the dot import feels like it should be avoided.

File: api/v1alpha1/job.go [hunk 2]
```diff
@@ -47,6 +47,12 @@ const (
 	Status       PowerAction = "status"
 )
 
+// Pointer provides an easy way to retrieve the power action as a pointer for use in job
```
Comment (displague) [inline api/v1alpha1/job.go:50]: `&` is `an easy way to retrieve the power action as a pointer`. :-)

Other than 9 characters in each of the tests below, what does this add?

Is a `Pointer()` conventional or fulfill some interface?

**Review by jacobweinstock — CHANGES_REQUESTED** [review]

File: api/v1alpha1/job.go (hunk 2 above)
Comment (chrisdoherty4) [reply api/v1alpha1/job.go:50]: You can't retrieve a pointer to a constant so you need to create a variable every time you want to set this. This is a convenient thing and I'm debating making it Ptr()

File: controllers/job_controller_test.go (hunk 1 above)
Comment (chrisdoherty4) [reply controllers/job_controller_test.go:8]: > The use of this assert library

I really think that code review point is clutching at straws and would strongly encourage ignoring it. The assumptions it makes for its rationale are inaccurate and its alternative is hard to justify as 'better' given in practice tests aren't as clear cut and lots of branch/predicate logic can be hard to parse.

Comment (chrisdoherty4) [reply controllers/job_controller_test.go:8]: > The dot import

I took another look at this as, usually, I'd agree. Similar to the above though, if applied broadly without consideration of context you end up being uncomplimentary to the style. Specifically, Ginkgo and Gomega are tools for BDD. BDD attempts to use natural language to model expectations. Applying the dot import guideline results in breaks when constructing assertions which defeats the point. 

I'm not fond of BDD and would prefer we remove this entirely - I might do just that on a separate PR. I'll update this given I intend on removing it anyway but I wanted to call it out nonetheless.

**Review by jacobweinstock — APPROVED** [review]
DCO needs to be signed.
